### DISCUSSION TITLE
Rock and a hard place: model_config clashes with an existing field

### DISCUSSION THREAD

**Discussion by geodavic** [pydantic/pydantic#10607] [Question]
I am using openapi-generator to generate pydantic code from swagger docs. in those docs is a model with a field called `model_config`. The openapi-generator then creates a class like:

```py
class Thing(BaseModel):
    field1: str
    field2: str
    model_config: Dict[str,Any]
    field4: str 
    ...


    model_config = ConfigDict( ... )
```

This is a problem because, obviously, the field `model_config` conflicts with the unrelated pydantic `model_config` dict. When I create an instance of this model, the `model_config` field doesn't exist. What can I do? I don't have control over the swagger docs, so changing the field name is not possible. Is there any way around this that doesn't involve manually editing the openapi-generated code?

**Comment by Viicos** [comment]
Using [aliases](https://docs.pydantic.dev/latest/concepts/alias/) might help.

**Reply by geodavic** [reply]
Unless I misunderstand, it looks like those are for serialization and validation. I was hoping there is a way to create an alias for a field that can be used to access that attribute from an instance. For example:

```py
class Thing(BaseModel):
    model_config: Dict[str,Any] = Field(field_alias="config")

    model_config = ConfigDict(validate_assignment=True)

thing = Thing(model_config={"a":1})
thing.config == thing.model_config # True
```

(I made up "field_alias" just to demonstrate my desired behavior).

**Reply by geodavic** [reply]
(of course I could add a `@property` method, but that's not a pydantic solution that my codegen tool is likely to support)
